CMakeLists.txt

```diff
@@ -243,13 +243,15 @@ endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
-find_package(LibCrypto REQUIRED)
-target_link_libraries(${PROJECT_NAME} PUBLIC LibCrypto::Crypto ${OS_LIBS} m)
+if (NOT IN_SOURCE_BUILD)
```

Comment:
As I'm looking at the changes you made to `FindLibCrypto.cmake`, do we even need this check at all? You're already handling `if (crypto_FOUND)` in there.

Reply:
`IN_SOURCE_BUILD` is not a standard CMake variable, it's only used in CRT libraries. For example, https://github.com/awslabs/aws-crt-cpp/blob/6513a25480077cb825813372701d15bae077c65a/CMakeLists.txt#L99
With `IN_SOURCE_BUILD`, we will just do `add_subdirectory`, instead of `find_package`. This allows us build everything together, rather than installing the dependencies first.